tests/api.c

```diff
@@ -5583,6 +5589,760 @@ static int test_wc_ChaCha20Poly1305_aead (void)
+        0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65
+    };
+    byte    badKey24[] =
+    {
+        0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37,
+        0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66,
+        0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36
+    };
+    byte   badKey32[] =
+    {
+        0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x37, 0x37,
+        0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66,
+        0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37,
+        0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65
+    };
+    int     ret;
+
+    printf(testingFmt, "wc_AesGcmSetKey()");
+
+    ret = wc_AesGcmSetKey(&aes, key16, sizeof(key16)/sizeof(byte));
+    if (ret == 0) {
+        ret = wc_AesGcmSetKey(&aes, key24, sizeof(key24)/sizeof(byte));
+    }
+    if (ret == 0) {
+        ret = wc_AesGcmSetKey(&aes, key32, sizeof(key32)/sizeof(byte));
+    }
+
+    /* Pass in bad args. */
+    if (ret == 0) {
+        ret = wc_AesGcmSetKey(&aes, badKey16, sizeof(badKey16)/sizeof(byte));
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_AesGcmSetKey(&aes, badKey24, sizeof(badKey24)/sizeof(byte));
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_AesGcmSetKey(&aes, badKey32, sizeof(badKey32)/sizeof(byte));
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = 0;
+        } else {
+            ret = SSL_FATAL_ERROR;
+        }
+    }
+
+    printf(resultFmt, ret == 0 ? passed : failed);
+
+#endif
+    return 0;
+} /* END test_wc_AesGcmSetKey */
+
+/*
+ * test function for wc_AesGcmEncrypt and wc_AesGcmDecrypt
+ */
+static int test_wc_AesGcmEncryptDecrypt (void)
+{
+#if !defined(NO_AES) && defined(HAVE_AESGCM)
+
+    Aes     aes;
+    byte    key32[] =
+    {
+        0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37,
+        0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66,
+        0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37,
+        0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66
+    };
+    byte    vector[] = /* Now is the time for all w/o trailing 0 */
+    {
+        0x4e,0x6f,0x77,0x20,0x69,0x73,0x20,0x74,
+        0x68,0x65,0x20,0x74,0x69,0x6d,0x65,0x20,
+        0x66,0x6f,0x72,0x20,0x61,0x6c,0x6c,0x20
+    };
+    const byte a[] =
+    {
+        0xfe, 0xed, 0xfa, 0xce, 0xde, 0xad, 0xbe, 0xef,
+        0xfe, 0xed, 0xfa, 0xce, 0xde, 0xad, 0xbe, 0xef,
+        0xab, 0xad, 0xda, 0xd2
+    };
+    byte    iv[]   = "1234567890a";
+    byte    badIV[]  = "1234567890abcde";
+    byte    enc[sizeof(vector)];
+    byte    resultT[AES_BLOCK_SIZE];
+    byte    dec[sizeof(vector)];
+    int     ret;
+    int     gcmD     =   SSL_FATAL_ERROR;
+    int     gcmE     =   SSL_FATAL_ERROR;
+
+    /* Init stack variables. */
+    XMEMSET(enc, 0, sizeof(vector));
+    XMEMSET(dec, 0, sizeof(vector));
+    XMEMSET(resultT, 0, AES_BLOCK_SIZE);
+
+    ret = wc_AesGcmSetKey(&aes, key32, sizeof(key32)/sizeof(byte));
+    if (ret == 0) {
+        ret = wc_AesGcmEncrypt(&aes, enc, vector, sizeof(vector),
```

Comment:
Should these calls to wc_AesGcmEncrypt/Decrypt() be after the printf for wc_AesGcmEncrypt?  Or was this intended?